src/reference/config/bindings/binding-openapi.md

```diff
@@ -0,0 +1,233 @@
+
+The `client` kind `openapi` binding creates composite of `tcp`, `tls`, and `http` bindings with server kind and adapts `openapi` request-response streams to `http` request-response streams.
+
+## Configuration
+
+:::: note Properties
+
+- [kind\*](#kind)
+- [options](#options)
+  - [options.tcp](#options-tcp)
+    - [tcp.host](#tcp-host)
+    - [tcp.port](#tcp-port)
+  - [options.tls](#options-tls)
+    - [tls.version](#tls-version)
+    - [tls.keys](#tls-keys)
+    - [tls.trust](#tls-trust)
+    - [tls.signers](#tls-signers)
+    - [tls.trustcacerts](#tls-trustcacerts)
+    - [tls.sni\*](#tls-sni)
+    - [tls.alpn](#tls-alpn)
+    - [tls.mutual](#tls-mutual)
+  - [options.http](#options-http)
+    - [options.authorization](#http-authorization)
+      - [authorization.credentials](#authorization-credentials)
+        - [credentials.cookies](#credentials-cookies)
+        - [credentials.headers](#credentials-headers)
+        - [credentials.query](#credentials-query)
+  - [options.specs](#options-specs)
+- [exit](#exit)
+
+::: right
+\* required
+:::
+
+::::
+
+### kind\*
+
+> `enum` [ "client", "server" ]
+
+Behave as a `openapi` `client` or `server`.
+
+```yaml
+kind: server
+```
+
+### options
+
+> `object`
+
+`openapi` - specific options.
+
+```yaml
+options:
+  spec:
+    openapi-id: spec/openapi.yaml
+```
+
+### options.tcp
+
+> `object`
+
+`client` specific `tcp` options.
+
+### tpc.host
+
+> `string`
+
+Hostname or IP address.
+
+### tcp.port
+
+> `integer` | `string` | `array` of  `integer` | `array` of `string`
+
+Port number(s), including port number ranges.
+
+### options.tls
+
+> `object`
+
+`tls` specific options.
+
+### tls.version
+
+> `string`
+
+Protocol version.
+
+### tls.keys
+
+> `array` of `string`
+
+A list of reference names for the Vault key.
+
+### tls.trust
+
+> `array` of `string`
+
+A list of reference names for the Vault certificate.
+
+### tls.signers
+
+> `array` of `string`
+
+A list of reference names for the Vault signer certificate.
+
+### tls.trustcacerts
+
+> `boolean` | Default: `true` when trust is `null`
+
+Trust CA certificates.
+
+### tls.sni\*
+
+> `array` of `string`
+
+A list of the Server Name Indications.
+
+### tls.alpn
+
+> `array` of `string`
+
+Application protocols.
+
+### tls.mutual
+
+> `enum` [ "required", "requested", "none" ] | Default: `"none"`
+
+Mutual authentication
+
+### options.http
+
+> `object`
+
+`http` specific options.
+
+### http.authorization
+
+> `object` as map of named properties
+
+Authorization by guard for the `HTTP/1.1` and `HTTP/2` protocols.
+
+```yaml
+authorization:
+  jwt:
+    credentials:
+      headers:
+        authorization: Bearer {credentials}
+```
+
+#### authorization.credentials
+
+> `object`
+
+Defines how to extract credentials from the HTTP request.
+
+##### credentials.cookies
+
+> `object` as map of `string`
```

Comment:
```suggestion
> `map` of `name: value` properties
```
I am working on how to define this type and came up with the above definition. I have updated the other docs with this but let me know if it makes more sense to you?